zetaclient/chains/bitcoin/tx_script.go

```diff
@@ -245,3 +275,44 @@ func DecodeTSSVout(vout btcjson.Vout, receiverExpected string, chain chains.Chai
 
 	return receiverVout, amount, nil
 }
+
+func decodeInscriptionPayload(t *scriptTokenizer) ([]byte, error) {
+	if !t.Next() || t.Opcode() != txscript.OP_FALSE {
+		return nil, fmt.Errorf("OP_FALSE not found")
+	}
+
+	if !t.Next() || t.Opcode() != txscript.OP_IF {
+		return nil, fmt.Errorf("OP_IF not found")
+	}
+
+	memo := make([]byte, 0)
```

Comment:
What is the maximum byte size a bitcoin message can carry?

Reply:
From what I know, taproot script txn size is lifted, so to play safe we assume the max block size, 4MB. But usually a few hundred KB is already very expensive.